src/windowing/window.cpp

```diff
@@ -0,0 +1,150 @@
+#include <glad/glad.h>
+
+// GLAD must be included first. clang-format sorts includes, hence this comment
+// here.
+
+#include <windowing/window.hpp>
+
+namespace nebula {
+
+  Window::Window(void (*custom_render_func)())
+  {
+    // Initialize GLFW
+    glfwInit();
+
+    // Get the primary monitor's video mode to determine the screen resolution
+    GLFWmonitor* primary_monitor = glfwGetPrimaryMonitor();
+    const GLFWvidmode* mode = glfwGetVideoMode(primary_monitor);
+    width = mode->width;
+    height = mode->height;
+    custom_render = custom_render_func;
+    currently_moved_rectangle = nullptr;
+    // Create a GLFW window
+    glfw_window = glfwCreateWindow(width, height, "Nebula", nullptr, nullptr);
+
+    if(!glfw_window) {
+      glfwTerminate();
+    }
+    // Set custom window pointer to this structure
+    glfwSetWindowUserPointer(glfw_window, this);
+
+    // Set callbacks
+    glfwSetMouseButtonCallback(glfw_window, mouse_button_callbacks);
+    glfwSetKeyCallback(glfw_window, keyboard_button_callback);
+    glfwSetCursorPosCallback(glfw_window, cursor_position_callbacks);
+  }
+
+  Window::~Window()
+  {
+    glfwDestroyWindow(glfw_window);
+    glfwTerminate();
+  }
+
+  /* Public methods */
+  u8 Window::make_current_context() const
```

Comment:
based on the name of the function, what does this function do? can't tell

Reply:
It's unnecessary tbh if we are using only one window. I'm moving this to the constructor.